pkg/openSearch/esextension/scriptextension.go

```diff
@@ -3,15 +3,30 @@ package esextensions
 // ... other imports
 import "github.com/aquasecurity/esquery"
 
+// ScriptedSumAggregation represents an aggregation that calculates the sum using a scripted expression.
 type ScriptedSumAggregation struct {
 	name   string
 	script string
 }
 
+// Name returns the name of the ScriptedSumAggregation.
+//
+// Example Usage:
+//
+//	a := ScriptedSumAgg("testName", "testScript")
+//	name := a.Name()
+//
+// Returns:
+//
+//	The name of the ScriptedSumAggregation.
 func (a *ScriptedSumAggregation) Name() string {
 	return a.name
 }
 
+// Map returns a map representation of the ScriptedSumAggregation.
+// The map includes a "sum" key with a corresponding value that contains another map.
```

Comment:
Do we really want to document it in such a detailled way? If the implementation only changes a bit, we must also remember to change the comment :-D
For me the first line would be totally sufficient!